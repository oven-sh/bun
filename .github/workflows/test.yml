name: Test

concurrency:
  group: ${{ github.workflow }}-${{ inputs.run-id || github.run_id }}
  cancel-in-progress: true

on:
  push:
  workflow_call:
    inputs:
      run-id:
        type: string
        required: true

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/download-artifact@v4
        with:
          run-id: ${{ inputs.run-id || '8623134412' }}
          pattern: bun-*
      - run: |
          ls -la
