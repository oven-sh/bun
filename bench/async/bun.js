import { bench, run } from "mitata";

bench("sync", () => {});
bench("async", async () => {});
bench("await 1", async () => await 1);

await run();
