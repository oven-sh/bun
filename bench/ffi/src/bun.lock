{
  "lockfileVersion": 0,
  "workspaces": {
    "": {
      "name": "bench",
      "dependencies": {
        "@napi-rs/cli": "^2.10.1",
        "@node-rs/helper": "^1.3.3",
      },
    },
  },
  "packages": {
    "@napi-rs/cli": ["@napi-rs/cli@2.14.8", "", { "bin": { "napi": "scripts/index.js" } }, "sha512-IvA3s8BqohMdUbOkFn7+23u1dhIJZCkA8Xps7DD4SLdCMbcbUF6MUuKiqxuqmVHBFTaxU25sU56WdX3efqGgPw=="],

    "@napi-rs/triples": ["@napi-rs/triples@1.1.0", "", {}, "sha512-XQr74QaLeMiqhStEhLn1im9EOMnkypp7MZOwQhGzqp2Weu5eQJbpPxWxixxlYRKWPOmJjsk6qYfYH9kq43yc2w=="],

    "@node-rs/helper": ["@node-rs/helper@1.3.3", "", { "dependencies": { "@napi-rs/triples": "^1.1.0" } }, "sha512-p4OdfQObGN9YFy5WZaGwlPYICQSe7xZYyXB0sxREmvj1HzGKp5bPg2PlfgfMZEfnjIA882B9ZrnagYzZihIwjA=="],
  }
}
