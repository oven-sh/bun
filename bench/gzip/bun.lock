{
  "lockfileVersion": 0,
  "workspaces": {
    "": {
      "name": "bench",
      "dependencies": {
        "@babel/standalone": "7.24.10",
      },
    },
  },
  "packages": {
    "@babel/standalone": ["@babel/standalone@7.24.10", "", {}, "sha512-nGC37EKfmelpyCXto1pw6SBkD5ZQRdMbL6WISi28xWit9dtiy9dChU1WgEfzturUTxrmOGkMDRrCydFMA7uOaQ=="],
  }
}
