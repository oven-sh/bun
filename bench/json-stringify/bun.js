import { bench, run } from "mitata";

bench("JSON.stringify({hello: 'world'})", () =>
  JSON.stringify({ hello: "world" }),
);

const otherUint8Array = new Uint8Array(1024);
bench("Uint8Array.from(otherUint8Array)", () =>
  Uint8Array.from(otherUint8Array),
);

run();
