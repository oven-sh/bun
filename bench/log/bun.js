import { bench, run } from "mitata";

bench("console.log('hello')", () => console.log("hello"));
bench("console.log({ hello: 'object' })", () => console.log({ hello: "object" }));
await run();
