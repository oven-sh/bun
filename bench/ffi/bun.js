import { ptr, dlopen, CString, toBuffer } from "bun:ffi";
import { run, bench, group } from "mitata";

const { napiNoop, napiHash, napiString } = require(import.meta.dir + "/src/ffi_napi_bench.node");

const {
  symbols: {
    ffi_noop: { native: ffi_noop },
    ffi_hash: { native: ffi_hash },
    ffi_string: { native: ffi_string },
  },
} = dlopen(import.meta.dir + "/src/ffi_napi_bench.node", {
  ffi_noop: { args: [], returns: "void" },
  ffi_string: { args: [], returns: "ptr" },
  ffi_hash: { args: ["ptr", "u32"], returns: "u32" },
});

const bytes = new Uint8Array(64);

group("bun:ffi", () => {
  bench("noop", () => ffi_noop());
  bench("hash", () => ffi_hash(ptr(bytes), bytes.byteLength));

  bench("c string", () => new CString(ffi_string()));
});

if (process.env.SHOW_NAPI)
  group("bun:napi", () => {
    bench("noop", () => napiNoop());
    bench("hash", () => napiHash(bytes));

    bench("string", () => napiString());
  });

await run();
