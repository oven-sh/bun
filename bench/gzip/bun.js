import { run, bench, group } from "mitata";
import { gzipSync, gunzipSync } from "bun";

const data = new TextEncoder().encode("Hello World!".repeat(9999));

const compressed = gzipSync(data);

const libraries = ["zlib"];
if (Bun.semver.satisfies(Bun.version, ">=1.1.21")) {
  libraries.push("libdeflate");
}

const options = { library: undefined };
const benchFn = (name, fn) => {
  if (libraries.length > 1) {
    group(name, () => {
      for (const library of libraries) {
        bench(library, () => {
          options.library = library;
          fn();
        });
      }
    });
  } else {
    bench(name, () => {
      fn();
    });
  }
};

benchFn(`roundtrip - "Hello World!".repeat(9999))`, () => {
  gunzipSync(gzipSync(data, options), options);
});

benchFn(`gzipSync("Hello World!".repeat(9999)))`, () => {
  gzipSync(data, options);
});

benchFn(`gunzipSync("Hello World!".repeat(9999)))`, () => {
  gunzipSync(compressed, options);
});

await run();
