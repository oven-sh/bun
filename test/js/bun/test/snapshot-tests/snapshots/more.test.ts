import { it, test, expect, describe } from "bun:test";
describe("d0", () => {
  test("snapshot serialize edgecases", () => {
    expect(1).toMatchSnapshot();
    expect("1\b2\n3\r4").toMatchSnapshot();
    expect("\r\n").toMatchSnapshot();
    expect("1\b2\n3\r\r\r\r\r\r\r\r\r\r\r\r4\v5\f6\t7\\\n\r\n\n\nr\nr\n").toMatchSnapshot();
    expect("1\b2\n3\r4\v5\f6\t7\\").toMatchSnapshot();
    expect("\r").toMatchSnapshot();
    expect("\n").toMatchSnapshot();
    expect("\\").toMatchSnapshot();
    expect("\v").toMatchSnapshot();
    expect("\f").toMatchSnapshot();
    expect("\t").toMatchSnapshot();
    expect("\b").toMatchSnapshot();
    expect("\b\t").toMatchSnapshot();

    expect(`hello sn
    apshot`).toMatchSnapshot();
    expect(new String()).toMatchSnapshot();
    expect(new String("")).toMatchSnapshot();

    expect({ a: { b: 1 } }).toEqual({ a: { b: 1 } });
    expect("\\\nexport with test name\n\n").toMatchSnapshot();

    expect(1).toMatchSnapshot();
    expect(1).toMatchSnapshot("one");
    expect(2).toMatchSnapshot();
    expect(3).toMatchSnapshot("one");
    expect("`````````\\``````\\`\\``````\\`````\\``\\\\`\\````````````").toMatchSnapshot();
    expect("`````````\\``````\\`\\``````\\`````\\``\\\\`\\````````````\\").toMatchSnapshot();
    expect("\\`````````\\``````\\`\\``````\\`````\\``\\\\`\\````````````").toMatchSnapshot();
    expect("\\`````````\\``````\\`\\``````\\`````\\``\\\\`\\````````````\\").toMatchSnapshot();
    expect("one t`wo `three").toMatchSnapshot();
    expect("one tw\\`o three").toMatchSnapshot();
    expect("\nexport[\\`hello snap'shot 2`] = `").toMatchSnapshot();
    expect("\nexport[`hello snapshot 2`] = `").toMatchSnapshot();
    expect("`hello snapshot3 \\``").toMatchSnapshot();
    expect("`hello snapshot4 \\`\\`").toMatchSnapshot();
    expect("\\`hello snapshot5 \\`\\`").toMatchSnapshot();
    expect({ a: 1, b: 2, c: 3 }).toMatchSnapshot("¾");
    expect({ a: 1, b: 2, c: 3 }).toMatchSnapshot("\uD83D\uDC04");
    expect({ a: "\uD83D\uDC04", b: "🐈" }).toMatchSnapshot("😃");
  });
});

describe("d0", () => {
  describe("d1", () => {
    test("t1", () => {
      expect("hello`snapshot\\").toEqual("hello`snapshot\\");
      expect("hello`snapshot\\").toMatchSnapshot();
    });
    test("t2", () => {
      expect("hey").toMatchSnapshot();
    });
  });
  test("t3", () => {
    expect("hello snapshot").toMatchSnapshot();
  });
  test("t4", () => {
    expect("hello`snapshot\\").toMatchSnapshot();
  });
});
