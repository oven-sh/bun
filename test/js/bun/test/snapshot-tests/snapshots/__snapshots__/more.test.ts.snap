// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`d0 d1 t1 1`] = `"hello\`snapshot\\"`;

exports[`d0 d1 t2 1`] = `"hey"`;

exports[`d0 snapshot serialize edgecases 1`] = `1`;

exports[`d0 snapshot serialize edgecases 2`] = `
"12
3
4"
`;

exports[`d0 snapshot serialize edgecases 3`] = `
"
"
`;

exports[`d0 snapshot serialize edgecases 4`] = `
"12
3











456	7\\



r
r
"
`;

exports[`d0 snapshot serialize edgecases 5`] = `
"12
3
456	7\\"
`;

exports[`d0 snapshot serialize edgecases 6`] = `
"
"
`;

exports[`d0 snapshot serialize edgecases 7`] = `
"
"
`;

exports[`d0 snapshot serialize edgecases 8`] = `"\\"`;

exports[`d0 snapshot serialize edgecases 9`] = `""`;

exports[`d0 snapshot serialize edgecases 10`] = `""`;

exports[`d0 snapshot serialize edgecases 11`] = `"	"`;

exports[`d0 snapshot serialize edgecases 12`] = `""`;

exports[`d0 snapshot serialize edgecases 13`] = `"	"`;

exports[`d0 snapshot serialize edgecases 14`] = `
"hello sn
    apshot"
`;

exports[`d0 snapshot serialize edgecases 15`] = `String {}`;

exports[`d0 snapshot serialize edgecases 16`] = `String {}`;

exports[`d0 snapshot serialize edgecases 17`] = `
"\\
export with test name

"
`;

exports[`d0 snapshot serialize edgecases 18`] = `1`;

exports[`d0 snapshot serialize edgecases 19`] = `2`;

exports[`d0 snapshot serialize edgecases 20`] = `"\`\`\`\`\`\`\`\`\`\\\`\`\`\`\`\`\\\`\\\`\`\`\`\`\`\\\`\`\`\`\`\\\`\`\\\\\`\\\`\`\`\`\`\`\`\`\`\`\`\`"`;

exports[`d0 snapshot serialize edgecases 21`] = `"\`\`\`\`\`\`\`\`\`\\\`\`\`\`\`\`\\\`\\\`\`\`\`\`\`\\\`\`\`\`\`\\\`\`\\\\\`\\\`\`\`\`\`\`\`\`\`\`\`\`\\"`;

exports[`d0 snapshot serialize edgecases 22`] = `"\\\`\`\`\`\`\`\`\`\`\\\`\`\`\`\`\`\\\`\\\`\`\`\`\`\`\\\`\`\`\`\`\\\`\`\\\\\`\\\`\`\`\`\`\`\`\`\`\`\`\`"`;

exports[`d0 snapshot serialize edgecases 23`] = `"\\\`\`\`\`\`\`\`\`\`\\\`\`\`\`\`\`\\\`\\\`\`\`\`\`\`\\\`\`\`\`\`\\\`\`\\\\\`\\\`\`\`\`\`\`\`\`\`\`\`\`\\"`;

exports[`d0 snapshot serialize edgecases 24`] = `"one t\`wo \`three"`;

exports[`d0 snapshot serialize edgecases 25`] = `"one tw\\\`o three"`;

exports[`d0 snapshot serialize edgecases 26`] = `
"
export[\\\`hello snap'shot 2\`] = \`"
`;

exports[`d0 snapshot serialize edgecases 27`] = `
"
export[\`hello snapshot 2\`] = \`"
`;

exports[`d0 snapshot serialize edgecases 28`] = `"\`hello snapshot3 \\\`\`"`;

exports[`d0 snapshot serialize edgecases 29`] = `"\`hello snapshot4 \\\`\\\`"`;

exports[`d0 snapshot serialize edgecases 30`] = `"\\\`hello snapshot5 \\\`\\\`"`;

exports[`d0 snapshot serialize edgecases: one 1`] = `1`;

exports[`d0 snapshot serialize edgecases: one 2`] = `3`;

exports[`d0 snapshot serialize edgecases: ¬æ 1`] = `
{
  "a": 1,
  "b": 2,
  "c": 3,
}
`;

exports[`d0 snapshot serialize edgecases: üêÑ 1`] = `
{
  "a": 1,
  "b": 2,
  "c": 3,
}
`;

exports[`d0 snapshot serialize edgecases: üòÉ 1`] = `
{
  "a": "üêÑ",
  "b": "üêà",
}
`;

exports[`d0 t3 1`] = `"hello snapshot"`;

exports[`d0 t4 1`] = `"hello\`snapshot\\"`;
